### PR TITLE
feat(protocol): Add Protocol.Extra.module_implements?/2

### PR DESCRIPTION
**Pull request by glittershark**
Add Protocol.Extra.module_implements?/2, a predicate for determining if
a given struct *module* implements a given protocol.

I've tested this particular implementation both with and without
protocol consolidation, and both cases seem to work properly.

### CODE REVIEW

File: lib/protocol/extra.ex [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+      iex> Protocol.Extra.module_implements?(Enumerable, Stream)
+      true
+
+      iex> Protocol.Extra.module_implements?(String.Chars, Stream)
+      false
+
+  """
+  @spec module_implements?(Protocol.t, module) :: boolean
+  def module_implements?(protocol, struct_module)
```
Comment (rschmukler) [inline lib/protocol/extra.ex]: Thoughts on renaming to impl_by_module? - seems like a better name since the protocol is the first argument

**Review by rschmukler — COMMENTED** [review]
See below for a question on naming

**Review by russmatney — APPROVED** [review]
clever! lgtm

File: lib/protocol/extra.ex (hunk 1 above)
Comment (glittershark) [reply lib/protocol/extra.ex]: 1. I dislike abbreviating in public names, so if anything it'd be "implemented_by_module"
2. passive voice is :-1:

**Comment by glittershark** [comment]
per discussion with @rschmukler in PM, swapped out the original function for `Protocol.Extra.module_impl_for`, to better mimic the semantics and naming of the stdlib.

**Comment by russmatney** [comment]
Nice change - looks good!

**Pull request merged**